src/config/include/config/Node.hpp

```diff
@@ -0,0 +1,670 @@
+        : _yaml{yaml}, _parent{parent}, _valid{valid}, _key{std::move(key)} {}
+
+    Node(const YAML::Node yaml, const Node* const parent, std::string key)
+        : Node{yaml, parent, yaml, std::move(key)} {}
+
+
+    template <typename O, typename... Args>
+    static constexpr bool isNodeConstructible() {
+        // TODO: test of constness
+        return !std::is_trivially_constructible_v<O> &&
+            (std::is_constructible_v<O, Node&, Args...> ||
+             std::is_constructible_v<O, const Node&, Args...>);
+    }
+
+    template <typename O,
+              typename... Args,
+              typename = std::enable_if_t<isNodeConstructible<O, Args...>()>>
+    std::optional<O> _maybeImpl(Args&&... args) const {
+        return std::make_optional<O>(*this, std::forward<Args>(args)...);
+    }
+
+    template <typename O,
+              typename... Args,
+              typename = std::enable_if_t<
+                  !isNodeConstructible<O, Args...>() &&
+                  std::is_same_v<O, typename NodeConvert<O>::type> &&
+                  std::is_same_v<O,
+                                 decltype(NodeConvert<O>::convert(std::declval<Node>(),
+                                                                  std::declval<Args>()...))>>,
+              typename = void>
+    std::optional<O> _maybeImpl(Args&&... args) const {
+        return std::make_optional<O>(NodeConvert<O>::convert(*this, std::forward<Args>(args)...));
+    }
+
+    template <
+        typename O,
+        typename... Args,
+        typename = std::enable_if_t<
+            !isNodeConstructible<O, Args...>() &&
+            std::is_same_v<decltype(YAML::convert<O>::encode(std::declval<O>())), YAML::Node>>,
+        typename = void,
+        typename = void>
+    std::optional<O> _maybeImpl(Args&&... args) const {
+        static_assert(sizeof...(args) == 0,
+                      "Cannot pass additional args when using built-in YAML conversion");
+        return std::make_optional<O>(_yaml.as<O>());
+    }
+
+    static YAML::Node parse(std::string);
+
+    template <typename K>
+    std::optional<const YAML::Node> yamlGet(const K& key) const {
+        if (!_valid) {
+            return std::nullopt;
+        }
+        const YAML::Node found = _yaml[key];
+        if (found) {
+            return std::make_optional<const YAML::Node>(found);
+        } else {
+            if (!_parent) {
+                return std::nullopt;
+            }
+            return _parent->yamlGet(key);
+        }
+    }
+
+    template <typename K>
+    const Node get(const K& key) const {
+        const std::string keyStr = v1::toString(key);
+        if (!_valid) {
+            return Node{YAML::Node{}, this, false, keyStr};
+        }
+        if constexpr (std::is_convertible_v<K, std::string>) {
+            if (key == "..") {
+                // this is...not the most succinct business-logic ever....
+                std::stringstream childKey;
```

Comment:
1. [style nit] Standardize on either using `std::stringstream` or `std::ostringstream` but not both in these files.
2. [nit] This key is actually an alias for the parent so calling it `childKey` is confusing.
3. The following would be a bit more compact if you're willing to use `+` to do one of the string concatenations.

   ```cpp
   if (!_parent) {
       return Node{YAML::Node{}, nullptr, false, _key + "/.."};
   }

   std::ostringstream parentKey;
   parentKey << _parent->_key << "/" << _key << "/..";
   return Node{_parent->_yaml, _parent->_parent, _parent->_valid, parentKey.str()};
   ```

Reply:
1. Good call changed to `stringstream`
2. Took your suggestion for (3) another good suggestion.